packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1156,6 +1161,225 @@ class EndpointIndex<TInterval extends ISerializableInterval> implements Interval
 	}
 }
 
+/**
+ * Collection of intervals.
+ *
+ * Provide additional APIs to support efficiently querying a collection of intervals whose endpoints fall within a specified range.
+ */
+export interface IEndpointInRangeIndex<TInterval extends ISerializableInterval>
+	extends IntervalIndex<TInterval> {
+	/**
+	 * @returns an array of all intervals contained in this collection whose endpoints locate in the range [start, end] (includes both ends)
+	 */
+	findIntervalsWithEndpointInRange(start: number, end: number);
+}
+
+/**
+ * Collection of intervals.
+ *
+ * Provide additional APIs to support efficiently querying a collection of intervals whose startpoints fall within a specified range.
+ */
+export interface IStartpointInRangeIndex<TInterval extends ISerializableInterval>
+	extends IntervalIndex<TInterval> {
+	/**
+	 * @returns an array of all intervals contained in this collection whose startpoints locate in the range [start, end] (includes both ends)
+	 */
+	findIntervalsWithStartpointInRange(start: number, end: number);
+}
+
+/**
+ * Represents an object that has comparison overrides for smaller and larger comparisons.
+ */
+const alwaysCompareSmaller = Symbol();
+const alwaysCompareLarger = Symbol();
+
+interface HasComparisonOverride {
+	[alwaysCompareSmaller]: boolean;
+	[alwaysCompareLarger]: boolean;
+}
+
+/**
+ * Compares two partial objects with comparison overrides.
+ * @returns A negative number if a should be considered smaller, a positive number if b should be considered smaller, or 0 if they are equal.
+ */
+function compareOverrideables(a: any, b: any): number {
```

Comment:
this doesn't handle the case where both intervals have `alwaysCompareSmaller` or `alwaysCompareLarger`. I think the comparison here will actually be a lot simpler if you implement the suggestion of using a single symbol with a numeric value (maybe name it `forceCompare` instead and it has value -1 or 1). That also prevents confusing things like "what happens if I make `{ [alwaysCompareLarger]: true, [alwaysCompareSmaller]: true }`